pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -274,29 +276,31 @@ func (h *mtuHandler) Stop(uninstall bool) error {
 	return nil
 }
 
-func (h *mtuHandler) forceMssClamping(endpoint *submV1.Endpoint) error {
-	defaultHostIface, err := netlinkAPI.GetDefaultGatewayInterface()
-	if err != nil {
-		return errors.Wrapf(err, "Unable to find the default interface on host")
-	}
+func (h *mtuHandler) forceMssClamping(endpoint *submV1.Endpoint, usrForceMssVal int) error {
+	if usrForceMssVal == 0 {
+		defaultHostIface, err := netlinkAPI.GetDefaultGatewayInterface()
+		if err != nil {
+			return errors.Wrapf(err, "Unable to find the default interface on host")
+		}
 
-	overHeadSize := maxIpsecOverhead
-	if endpoint.Spec.Backend == vxlan.CableDriverName {
-		overHeadSize = vxlan.VxlanOverhead
-	}
+		overHeadSize := maxIpsecOverhead
+		if endpoint.Spec.Backend == vxlan.CableDriverName {
+			overHeadSize = vxlan.VxlanOverhead
+		}
 
-	klog.Infof("forceMssClamping to: IF_MTU(%d)-Overhead(%d)=%d",
-		defaultHostIface.MTU, overHeadSize, (defaultHostIface.MTU - overHeadSize))
+		usrForceMssVal = defaultHostIface.MTU - overHeadSize
+	}
 
+	klog.Infof("forceMssClamping to: %d", usrForceMssVal)
 	ruleSpecSource := []string{
 		"-m", "set", "--match-set", constants.LocalCIDRIPSet, "src", "-m", "set", "--match-set",
 		constants.RemoteCIDRIPSet, "dst", "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j", "TCPMSS",
-		"--set-mss", strconv.Itoa(defaultHostIface.MTU - overHeadSize),
+		"--set-mss", strconv.Itoa(usrForceMssVal),
 	}
 	ruleSpecDest := []string{
 		"-m", "set", "--match-set", constants.RemoteCIDRIPSet, "src", "-m", "set", "--match-set",
 		constants.LocalCIDRIPSet, "dst", "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j", "TCPMSS",
-		"--set-mss", strconv.Itoa(defaultHostIface.MTU - overHeadSize),
+		"--set-mss", strconv.Itoa(usrForceMssVal),
```

Comment:
It is very likely that user might initially deploy Submariner, notice the problem and then want to update the tcpMss value by adding the necessary annotation on the GW node. So the code here should first delete the previous iptable rule (if present) and create a new one with the usrForcedMssValue

Reply:
Not sure that I fully understand your comment, I assume that in case the user didn't annotate node we still want to clamp MSS to default value .
I think that latest version of PR should address your concern.